xds/src/main/java/io/grpc/xds/Bootstrapper.java

```diff
@@ -47,21 +47,28 @@
 
   private static final String LOG_PREFIX = "xds-bootstrap";
   private static final String BOOTSTRAP_PATH_SYS_ENV_VAR = "GRPC_XDS_BOOTSTRAP";
+  private static final String BOOTSTRAP_PATH_SYS_PROPERTY_VAR = "io.grpc.xds.bootstrap";
   @VisibleForTesting
   static final String CLIENT_FEATURE_DISABLE_OVERPROVISIONING =
       "envoy.lb.does_not_support_overprovisioning";
 
   private static final Bootstrapper DEFAULT_INSTANCE = new Bootstrapper() {
     @Override
     public BootstrapInfo readBootstrap() throws XdsInitializationException {
+      XdsLogger logger = XdsLogger.withPrefix(LOG_PREFIX);
       String filePath = System.getenv(BOOTSTRAP_PATH_SYS_ENV_VAR);
       if (filePath == null) {
-        throw new XdsInitializationException(
-            "Environment variable " + BOOTSTRAP_PATH_SYS_ENV_VAR + " not defined.");
+        filePath = System.getProperty(BOOTSTRAP_PATH_SYS_PROPERTY_VAR);
+        if (filePath == null) {
+          throw new XdsInitializationException(
+                  "Environment variable " + BOOTSTRAP_PATH_SYS_ENV_VAR
+                  + " or Java System Property " + BOOTSTRAP_PATH_SYS_PROPERTY_VAR
+                  + " not defined.");
+        }
+        logger.log(XdsLogLevel.INFO, BOOTSTRAP_PATH_SYS_PROPERTY_VAR + "={0}", filePath);
+      } else {
+        logger.log(XdsLogLevel.INFO, BOOTSTRAP_PATH_SYS_ENV_VAR + "={0}", filePath);
       }
-      XdsLogger
```

Comment:
Technically it's not necessary to remove this and use the `XdsLogger logger` variable. You can just replace the hardcoded string `BOOTSTRAP_PATH_SYS_ENV_VAR + "={0}"` with a String variable which is set to either `BOOTSTRAP_PATH_SYS_PROPERTY_VAR + "={0}"` or `BOOTSTRAP_PATH_SYS_ENV_VAR + "={0}"` in your 'if-else` statement. But this also works...

Reply:
Hmm, I did it like this to avoid having a temporary variable only used for the log message but maybe that's fine in this case. What about something like this?
```
String filePathSource = BOOTSTRAP_PATH_SYS_ENV_VAR;
String filePath = System.getenv(filePathSource);
if(filePath == null) {
  filePathSource = BOOTSTRAP_PATH_SYS_PROPERTY_VAR;
  filePath = System.getProperty(filePathSource);
}
if(filePath == null) {
  throw new XdsInitializationException(
         "Environment variable " + BOOTSTRAP_PATH_SYS_ENV_VAR
         + " or Java System Property " + BOOTSTRAP_PATH_SYS_PROPERTY_VAR
         + " not defined.");
} 
XdsLogger
   .withPrefix(LOG_PREFIX)
   .log(XdsLogLevel.INFO, filePathSource + "={0}", filePath);
```

Reply:
Sounds good. It's fine even if you don't change it since the logic is correct